pkg/container/vector/vector.go

```diff
@@ -632,6 +632,48 @@ func (v *Vector) PreExtend(rows int, mp *mpool.MPool) error {
 	return extend(v, rows, mp)
 }
 
+// PreExtendArea use to expand the mpool and area of vector
+func (v *Vector) PreExtendArea(rows int, mp *mpool.MPool) error {
```

Comment:
Why don't you add a arg, areaSize so that other types can also use it?

Reply:
PreExtendArea has same arguments as PreExtend
```go
func (v *Vector) PreExtend(rows int, mp *mpool.MPool) error {

```

And, sizes of other types are handled by this logic in PreExtendArea
```go
// get the size required for storing new rows
	var vSize int
	switch v.typ.Oid {
	case types.T_array_float32:
		vSize = 4 * int(v.typ.Width)
	case types.T_array_float64:
		vSize = 8 * int(v.typ.Width)
	default:
		vSize = v.typ.TypeSize()
	}
	vlen := vSize * rows
```

Reply:
Seems to me, that this computation does not belong to PreExtendArea, it is really outside, compute, pass in the result.    We want PreExtendArea can be used by vector of any Varlena type.